modules/shared/src/main/scala/org/tessellation/merkletree/MerkleTree.scala

```diff
@@ -0,0 +1,128 @@
+package org.tessellation.merkletree
+
+import cats.data.NonEmptyList
+import cats.kernel.Eq
+import cats.syntax.eq._
+import cats.syntax.option._
+
+import org.tessellation.security.hash.Hash
+
+import eu.timepit.refined.cats._
+import eu.timepit.refined.types.numeric.NonNegInt
+
+case class ProofEntry(target: Hash, siblings: Either[Hash, Hash])
```

Comment:
I think it would be beneficial to add a scala documentation for these classes/functions explaining the context and intended functionality - given that it is implemented with the base constructs (like e.g. fetching index from a list, left/right) which have some implicit meaning here.

Reply:
actually it would mean we have to explain how merkle trees should look like. does it make sense to just leave some url in the comment?